src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -36,25 +35,19 @@ public static VisualStudioInstance GetInstance(string workingDirectory)
                 return null;
             }
 
-            string versionContent = File.ReadAllText(versionPath);
-            Match versionMatch = VersionRegex.Match(versionContent);
-
-            if (!versionMatch.Success)
-            {
-                return null;
-            }
-
-            if (!int.TryParse(versionMatch.Groups[1].Value, out int major) ||
-                !int.TryParse(versionMatch.Groups[2].Value, out int minor) ||
-                !int.TryParse(versionMatch.Groups[3].Value, out int patch))
+            // Preview versions contain a hyphen after the numeric part of the version. Version.TryParse doesn't accept that.
+            string parseableVersion = File.ReadAllText(versionPath);
+            int indexOfHyphen = parseableVersion.IndexOf('-');
+            parseableVersion = indexOfHyphen >= 0 ? parseableVersion.Substring(0, indexOfHyphen) : parseableVersion;
+            if (!Version.TryParse(parseableVersion, out Version version) || version > Environment.Version)
```

Comment:
Should we first merge #79 and then move the version filtering to `RegisterDefaults`? I.e. the **default** behavior would be to pick the latest SDK not newer than the currently running runtime but the developer can implement their own policy if they choose so? There are legitimate use cases where the developer wants to know about all instances even if they are not compatible with the current process -- they might want to launch MSBuild out-of-proc for example.

Reply:
I'd like to get a minimal 5.0 fix out ASAP but I think this is a good way to go long-term.

Reply:
I made a change that I think implements this. I tested it with 5.0 and 3.1 runtimes, and they correctly identified 5.0 and 3.1 as the SDKs to use respectively, both ignoring my 2.2 SDK.